controller/src/beerocks/bml/internal/bml_internal.h

```diff
@@ -55,6 +56,40 @@ class bml_internal : public beerocks::socket_thread {
     int set_wifi_credentials(const std::string ssid, const std::string pass, int sec, int vap_id,
                              int force);
 
+    /**
+     * @brief set wifi credentials for specific AL-MAC
+     * 
+     * Sets wifi credentials for specific al_mac and sends autoconfig renew if requested.
+     *
+     * @param credentials credentials variable.
+     * @param al_mac al_mac of the agent to configure
+     * @return BML_RET_OK if successful, error code otherwise 
+     */
+    int set_wifi_credentials_new(const son::wireless_utils::sBssInfoConf &credentials,
+                                 sMacAddr &al_mac);
+
+    /**
+     * @brief clear wifi credentials for specific AL-MAC
+     * 
+     * Clears wifi credentials for specific al_mac.
+     *
+     * @param al_mac al_mac of the agent to configure
+     * @return BML_RET_OK if successful, error code otherwise 
+     */
+    int clear_wifi_credentials(sMacAddr &al_mac);
+
+    /**
+     * @brief update wifi credentials for specific AL-MAC
+     * 
+     * Sets wifi credentials for specific al_mac and sends autoconfig renew if requested.
```

Comment:
```suggestion
     * Sends autoconfig renew if credentials have changed for AL-MAC.
```

Note that for now, we'll simply *always* send autoconfig renew, even if credentials haven't changed. We can add a TODO somewhere to fix that.